### DISCUSSION TITLE
Order of owners in codeowners file

### DISCUSSION THREAD

**Discussion by DenizTC** [community/community#23644] [New to GitHub]
<p>I have my codeowners file set up and grouped by teams, so something like this:</p>
<pre><code class="lang-auto"># Team A
/engn/headers/sql_file1.in @ORG/team-a
/engn/headers/sql_file2.in @ORG/team-a
/engn/headers/sql_file3.in @ORG/team-a
/engn/headers/pdSql_file1.in @ORG/team-a

#Team B
/engn/headers/sqp_file1.in @ORG/team-b
/engn/headers/sqp_file2.in @ORG/team-b
/engn/headers/sqp_file3.in @ORG/team-b
/engn/headers/pd* @ORG/team-b
</code></pre>
<p>Who will be the owner of  <code>pdSql_file1.in</code> ? The <a href="https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners" rel="noopener nofollow ugc">docs</a> say the order matters, so I would assume since team-b comes after team-a, it’s general ownership overwrites the specific ownership of team-a.</p>
<p>Note I don’t want teams sharing the ownership of that file, it should be owned solely by team-a.</p>
<p>Is grouping by teams not the correct thing to do?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello <a class="mention" href="/deniztc">@DenizTC</a> and welcome to the community.</p>
<p>Yes, order matters in the <code>CODEOWNERS</code> file, specifically lines that come later override lines that appear earlier in the file. In the example in <a href="https://docs.github.com/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax">the documentation</a>, this is taken advantage of by starting with the least specific rule and then following with increasingly more specific rules as in this excerpt:</p>
<pre><code class="lang-auto"># These owners will be the default owners for everything in
# the repo. Unless a later match takes precedence,
# @global-owner1 and @global-owner2 will be requested for
# review when someone opens a pull request.
*       @global-owner1 @global-owner2

# Order is important; the last matching pattern takes the most
# precedence. When someone opens a pull request that only
# modifies JS files, only @js-owner and not the global
# owner(s) will be requested for a review.
*.js    @js-owner
</code></pre>
<aside class="quote no-group" data-username="DenizTC" data-post="1" data-topic="143073">
<div class="title">
<div class="quote-controls"></div>
 DenizTC:</div>
<blockquote>
<p>Is grouping by teams not the correct thing to do?</p>
</blockquote>
</aside>
<p>It isn’t specifically <em>incorrect</em> but it may mean that you’ll have to write more rules to achieve the same effect than if you worked with the way the <code>CODEOWNERS</code> file is interpreted. So you may want to reorder your file like this to achieve your stated goal:</p>
<pre><code class="lang-auto"># PD files
/engn/headers/pd* @ORG/team-b

# PD exceptions
/engn/headers/pdSql_file1.in @ORG/team-a

# etc ...
</code></pre>
